mlrun/utils/helpers.py

```diff
@@ -1335,6 +1335,10 @@ def datetime_from_iso(time_str: str) -> Optional[datetime]:
 def datetime_to_iso(time_obj: Optional[datetime]) -> Optional[str]:
     if not time_obj:
         return
+
+    # if the datetime object has a timezone, convert it to UTC before formatting
+    if time_obj.tzinfo:
+        time_obj = time_obj.astimezone(timezone.utc)
```

Comment:
We can't just override given timezone as it has meaning from user pov.
what you want to do is align user input with how it is stored on backend. is that correct? or perhaps the time obj didnt have timezone and you wanted to ensure tz (therefor the if condition is wrong)
anyway, what should be done is you need to align the datetime on the backend side, not client side. as you have UI as client as well sending requests with different timezones.